Server/mods/deathmatch/logic/CAccount.cpp

```diff
@@ -255,6 +255,15 @@ bool CAccount::IsIpAuthorized(const SString& strIp)
     return false;
 }
 
+//
+// Check if the serial has 32 hexadecimal characters
+//
+bool CAccount::IsValidSerial(const std::string& serial) noexcept
+{
+    const std::regex serialPattern("^[A-Fa-f0-9]{32}$");
+    return std::regex_match(serial, serialPattern);
```

Comment:
this function can throw `std::regex_error`. Please wrap this regex_match in try-catch:
```cpp
bool CAccount::IsValidSerial(const std::string& serial) noexcept
{
    const std::regex serialPattern("^[A-Fa-f0-9]{32}$");
    try {
        return std::regex_match(serial, serialPattern);
    } catch (...) {
        return false;
    }
}
```

Reply:
adjusted 9a14e49